### ISSUE TITLE
ROVER?: Compass Calibration silently fails if enabled compass is not present

### ISSUE THREAD

**Issue by jmachuca77** [ArduPilot/ardupilot#13257]
### Bug report

**Issue details**
When using only a UAVCAN compass (USE_COMPASS 0, USE_COMPASS2 0, USE_COMPASS3 1, COMPASS_PRIMARY 3), and the compass is not detected, Mag Calibration does not start, but there is no indication anything failed.

**Version**
ROVER 4.0.0 release 

**Platform**
Not sure if this is only on Rover, might be on other platforms too.
[  ] All
[  ] AntennaTracker
[  ? ] Copter
[ ? ] Plane
[ X ] Rover
[ ? ] Submarine

**Hardware type**
Tested using Here2 on UAVCAN with CubeBlack and CubeOrange

**Comment by jmachuca77** [comment]
Upon further testing this happens also if the only compass enabled is an external compass. Even if it is on I2C and it is correctly detected.

**Comment by jmachuca77** [comment]
This TLOG shows an attempt to start compass calibration with only the external compass enabled.

[2020-01-13 12-24-43.tlog.zip](https://github.com/ArduPilot/ardupilot/files/4055388/2020-01-13.12-24-43.tlog.zip)

**Comment by WickedShell** [comment]
We just need to send a NACK here: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Compass/AP_Compass_Calibration.cpp#L116 if no calibrations were started, as well as NACK here: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Compass/AP_Compass_Calibration.cpp#L99

**Comment by jmachuca77** [comment]
This TLOG shows how after enabling an internal compass and starting calibration it starts correctly.

[2020-01-13 12-29-13.tlog.zip](https://github.com/ArduPilot/ardupilot/files/4055433/2020-01-13.12-29-13.tlog.zip)

**Comment by tridge** [comment]
note that in these logs COMPASS_ENABLE=0

**Comment by rmackay9** [comment]
*maybe* we don't display a message to the user when they try to do a compass calibration when compass-enable = 0.

**Comment by TharakaUJ** [comment]
> _maybe_ we don't display a message to the user when they try to do a compass calibration when compass-enable = 0.

Yes, I tested this in SITL (Rover) and was able to reproduce the behavior.

When `COMPASS_ENABLE=0` or when calibration cannot start, `MAV_CMD_DO_START_MAG_CAL` returns `FAILED`, but there is no explanatory message sent to the GCS. In MAVProxy/QGC the user only sees the command failure without knowing why calibration did not start.

I verified this using both MAVProxy (`magcal start`) and QGroundControl.

If adding a message is what needed to close this issue, I am happy to create a PR.

**Comment by rmackay9** [comment]
Hi @TharakaUJ,

It's an old issue but sure, a PR would be good, txs!
